@damon-atkins 
This is what I get on OpenBSD, which was to be expected:
```
Traceback (most recent call last):
  File "test.py", line 30, in <module>
    test()
  File "test.py", line 20, in test
    ut_ret['seconds'] =  int(time.time()-time.mktime(utc_time.timetuple()))
UnboundLocalError: local variable 'utc_time' referenced before assignment
```

--- a/salt/modules/status.py
+++ b/salt/modules/status.py
@@ -140,6 +140,7 @@ def uptime():
         key/value pairs containing uptime information, instead of the output
         from a ``cmd.run`` call.
     .. versionchanged:: carbon
+        Support for OpenBSD and Solaris-like platforms.
         Fall back to output of `uptime` when /proc/uptime is not available.
 
     CLI Example:
@@ -162,6 +163,12 @@ def uptime():
             raise CommandExecutionError('The boot_time kstat was not found.')
         utc_time = datetime.datetime.utcfromtimestamp(float(res['stdout'].split()[1].strip()))
         ut_ret['seconds'] = int(time.time()-time.mktime(utc_time.timetuple()))
+    elif salt.utils.is_openbsd():
+        res = __salt__['cmd.run_all']('sysctl -n kern.boottime')
+        if res['retcode'] > 0:
+            raise CommandExecutionError('The sysctl kern.boottime was not found.')
+        utc_time = datetime.datetime.utcfromtimestamp(float(res['stdout'].strip()))
+        ut_ret['seconds'] = int(time.time()-time.mktime(utc_time.timetuple()))
     elif salt.utils.which('uptime'):
         return __salt__['cmd.run']('uptime')
     else:
@@ -173,7 +180,10 @@ def uptime():
     hours = (ut_ret['seconds'] - (ut_ret['days'] * 24 * 60 * 60)) // 60 // 60
     minutes = ((ut_ret['seconds'] - (ut_ret['days'] * 24 * 60 * 60)) // 60) - hours * 60
     ut_ret['time'] = '{0}:{1}'.format(hours, minutes)
-    ut_ret['users'] = len(__salt__['cmd.run']("who -s").split(os.linesep))
+    if salt.utils.is_openbsd():
+        ut_ret['users'] = len(__salt__['cmd.run']("who").split(os.linesep))
+    else:
+        ut_ret['users'] = len(__salt__['cmd.run']("who -s").split(os.linesep))
 
     return ut_ret
 

--- a/salt/modules/status.py
+++ b/salt/modules/status.py
@@ -161,7 +161,7 @@ def uptime():
         if res['retcode'] > 0:
             raise CommandExecutionError('The boot_time kstat was not found.')
         utc_time = datetime.datetime.utcfromtimestamp(float(res['stdout'].split()[1].strip()))
-        ut_ret['seconds'] = int((datetime.datetime.utcnow() - utc_time).total_seconds())
+        ut_ret['seconds'] = int(time.time()-time.mktime(utc_time.timetuple()))
     elif salt.utils.which('uptime'):
         return __salt__['cmd.run']('uptime')
     else:

--- a/salt/modules/status.py
+++ b/salt/modules/status.py
@@ -140,8 +140,7 @@ def uptime():
         key/value pairs containing uptime information, instead of the output
         from a ``cmd.run`` call.
     .. versionchanged:: carbon
-        Support for OpenBSD and Solaris-like platforms.
-        Fall back to output of `uptime` when /proc/uptime is not available.
+        Support for OpenBSD, FreeBSD, and Solaris-like platforms.
 
     CLI Example:
 
@@ -169,8 +168,16 @@ def uptime():
             raise CommandExecutionError('The sysctl kern.boottime was not found.')
         utc_time = datetime.datetime.utcfromtimestamp(float(res['stdout'].strip()))
         ut_ret['seconds'] = int(time.time()-time.mktime(utc_time.timetuple()))
-    elif salt.utils.which('uptime'):
-        return __salt__['cmd.run']('uptime')
+    elif salt.utils.is_freebsd():
+        res = __salt__['cmd.run_all']('sysctl -n kern.boottime')
+        if res['retcode'] > 0:
+            raise CommandExecutionError('The sysctl kern.boottime was not found.')
+        # format: { sec = 1477761334, usec = 664698 } Sat Oct 29 17:15:34 2016
+        # note: sec -> unixtimestamp
+        utc_time = datetime.datetime.utcfromtimestamp(
+            float(res['stdout'][1:res['stdout'].index(',')].split('=')[1].strip())
+        )
+        ut_ret['seconds'] = int(time.time()-time.mktime(utc_time.timetuple()))
     else:
         raise CommandExecutionError('This platform is not supported')
 

